woocommerce/woocommerce-android
# Adds a UI test that checks all reviews content.

### Description
This PR introduces a UI test that checks the content of the reviews list, and every individual review screen. It uses [Espresso](https://developer.android.com/training/testing/espresso/basics) matches to build assertions.

Please note that out of 23 changed files, 17 are WireMock files that were missing. WireMock has a [read-only](https://github.com/wiremock/wiremock/blob/01178a8b2719c7633beebd3a4e3e7ea3b95a89eb/src/main/java/com/github/tomakehurst/wiremock/junit/WireMockRule.java#L36) 🤯 `failOnUnmatchedRequests` [setting](https://github.com/wiremock/wiremock/issues/795#issuecomment-343313593), with default being `true`, that will force the test to fail in case of even a single unmatched request. So, while the test was technically passing (all assertions passed) without the added JSONs, I still had to add them 😢 .

More on the test specifics in the additional code comments.

### Testing instructions

1. Make sure you use the latest revision of mobile secrets
2. Use the gradle.properties from mobile secrets
3. To avoid possible Emulator flakiness, create a fresh Emulator instance. I tested under Pixel 4a API 30
4. Run `reviewListShowsAllReviews` test from `ReviewsUITest.kt` ~> it should pass

<img width="604" alt="Screenshot 2021-10-22 at 15 28 23" src="https://user-images.githubusercontent.com/73365754/138455723-76d2549b-b018-4c4d-b6d3-9fdaa7755a1e.png">

### Possible issues
I'm not sure how this will behave on the setups with lower performance. So far, I'm not using any conditional or simple sleeps.
